File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size())
            return false;
        
        map<char, int> m_s;
        map<char, int> m_t;

        for (char ch : s) {
            m_s[ch]++;
        }
        for (char ch : t) {
            m_t[ch]++;
        }

        return m_t == m_s;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        Map<Character, Integer> map = new HashMap<>();
        for (char c : s.toCharArray()) {
            map.put(c, map.getOrDefault(c, 0) + 1);
        }

        for (char c : t.toCharArray()) {
            int cnt = map.getOrDefault(c, 0);
            if (cnt == 0) {
                return false;
            }
            map.put(c, cnt - 1);
        }
        return false;
    }
}